### PR TITLE
Binarize adaptive

### PR DESCRIPTION
**Pull request by andrewluetgers**
Adds Wolf and Nick binarization algorithms. both are locally adaptive algorithms and for many documents are a great improvement over the global optimization approach of Otsu. 

This implementation is derived from [the source Christian Wolf provided](http://liris.cnrs.fr/christian.wolf/software/binarize/) along with his original publication. His implementation is very efficient as it made use of integral images to speed up the moving window calculations that are the core of the Niblack algorithm both of these algorithms are based on. Subsequently I added the Nick algorithm to his code. I also added a handy mat8ToPix to complement the existing pix8ToMat in image.cc.

For a good comparison of these binarization techniques I point you to the excellent [Nick paper](https://ai2-s2-pdfs.s3.amazonaws.com/76bd/5997bdd6d0a2611c590f5cf30f95c88e18ad.pdf).

In short these algorithms performs admirably in situations such as locally shaded areas, bleed-through or stains where Otsu's global threshold would fail. The arguments are window width, height and a K parameter usually 0.5 for Wolf and between -0.1 and -0.2 for Nick.

Nick will often catch faint and thin features that Wolf will miss. Wolf can be less noisy. Both are often lighter and thinner than Otsu but then again also much more usable in many cases.

compare

### CODE REVIEW

**Comment by andrewluetgers** [comment]
Should build fine on mac once https://github.com/creatale/node-dv/pull/32 is merged
